gradunwarp/core/coeffs.py

```diff
@@ -44,19 +43,16 @@ def coef_file_parse(cfile, txt_var_map):
 
     modifies txt_var_map in place
     '''
-    # parse .coef file. Strip unneeded characters. a valid line in that file is
-    # broken into validline_list
-    coef_re = re.compile(r'^[^\#]')  # regex for first character not a '#'
-    coef_file = open(cfile, 'r')
-    for line in coef_file.readlines():
-        if coef_re.match(line):
-            validline_list = line.lstrip(' \t').rstrip(';\n').split()
-            if validline_list:
-                log.info('Parsed : %s' % validline_list)
-                l = validline_list
-                x = int(l[1])
-                y = int(l[2])
-                txt_var_map[l[0]][x, y] = float(l[3])
+    with open(cfile) as coef_file:
```

Comment:
The main thing here is `coef_file` not getting closed except by the garbage collector. Probably fine, but occasionally you'll find a situation where you get noisy warnings if a filehandle is still open at process shutdown. This just closes it at the end of the block.

Reply:
Technically, this also changes the line parsing condition from "split returned at least one group of non-whitespace characters" to "some characters exist after removing specific characters from each end of the line", though it isn't clear what minimum silliness is required to make these behave differently (it may require nbsp or other exotic whitespace, which are unlikely to be present).

Reply:
Oh, you're right. I thought " ".split() would be `[" "]` for some reason. Let me revert that.